app/src/app/components/providers/UserProvider.tsx

```diff
@@ -0,0 +1,44 @@
+import { jwtDecode } from "jwt-decode";
+import { ReactNode, createContext, useContext, useState } from "react";
+import { getUserFromLocalStorage } from "../../../lib/localStorageManager";
+import { LocalStorageKeys } from "../../../lib/types/LocalStorageKeys";
+import { PacketJwtPayload } from "../../../types";
+import { UserProviderState, UserState } from "./types/UserTypes";
+
+const UserContext = createContext<UserProviderState | undefined>(undefined);
+
+export const useUser = () => {
+  const context = useContext(UserContext);
+
+  if (context === undefined) throw new Error("useUser must be used within a UserProvider");
```

Comment:
its just a general thing to ensure no one uses it outside provider. THe reason it is not used in auth config is because that component actually fetches data and can be in an error state,, thus cant do this error handling